### ISSUE TITLE
about res50

### ISSUE THREAD

**Issue by yanfangfangfang** [gjy3035/C-3-Framework#10]
thanks for you code ,i want to run model of res50,SHHB got result of  mae 8.3 and 13.8 with epoch 400 .but SHHA mae 109.5 mse 182.7 [
![image](https://user-images.githubusercontent.com/47957488/55784206-a8a4eb80-5ae2-11e9-9e43-27ad5c7170b0.png)
](url)
i found SHHB train with batch size=6 and because of different image size SHHA train with batch size=1 .res50 with some batch norm layer ,maybe question was occured here?  for comparsion i run SHHB with batch size=1 got the result of mae 17.0 mse 36
only one sample with batch_norm layer may affect result .but i don t have a better idea to deal with .could you give me some advices.

**Comment by Ethanwl** [comment]
You can also train SHHA with batch size=6.
The function SHHA_collate will make sure that the size in one batch is the same.

**Comment by gjy3035** [comment]
Sorry, recently, I have no time to implement Res50 on Shanghai A. 

For the multi-batch training, in addition to the suggestion of @Ethanwl , you also adopt the padding strategies to process the images with different sizes. The concrete code is available in another repo: [GCC-SFCN](https://github.com/gjy3035/GCC-SFCN/blob/master/datasets/UCF_QNRF.py).

According to the curves,  the model may be suffered from over fitting or it may performed not well in the congested regions.
I have some suggestions that you can try:
1) the learning rate is too large. You need to elaborately design a decay scheme and select the initialization of learning rate;
2) for mse loss function, you try to set `size_average=False` to aid net to focus on high-error regions. (simultaneously, the learning rate need to reset).

**Comment by yanfangfangfang** [comment]
very thanks for your reply

**Comment by gjy3035** [comment]
我们计划提供一个基于res50为backbone的网络，其在上海B上的结果要远远好于Res50的结果。上海A上的MAE也已经逼近60（其训练参数与Res50相差无几，可作为你的参考）。由于最近在准备其他事宜，后续完整检查完代码，我们会放出来（预计7、8月份）。多谢关注。

**Comment by clw5180** [comment]
> 我们计划提供一个基于res50为backbone的网络，其在上海B上的结果要远远好于Res50的结果。上海A上的MAE也已经逼近60（其训练参数与Res50相差无几，可作为你的参考）。由于最近在准备其他事宜，后续完整检查完代码，我们会放出来（预计7、8月份）。多谢关注。

请教下前辈，这个基于res50的模型是否已经开源
